src/Vipps.net/VippsApi.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Net.Http;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Vipps.net.Infrastructure;
+using Vipps.net.Services;
+
+namespace Vipps.net
+{
+    public interface IVippsApi
+    {
+        IVippsAccessTokenService AccessTokenService();
+        IVippsEpaymentService EpaymentService();
+        IVippsCheckoutService CheckoutService();
+    }
+
+    public class VippsApi : IVippsApi
+    {
+        private readonly VippsConfigurationOptions _vippsConfigurationOptions;
+        private VippsHttpClient _vippsHttpClient;
+        private ILoggerFactory _loggerFactory;
+        private readonly VippsAccessTokenService _accessTokenService;
+        private readonly IVippsEpaymentService _epaymentService;
+        private readonly IVippsCheckoutService _checkoutService; 
+        
+        public VippsApi(
```

Comment:
Should provide a way to inject a custom HttpClient (if we want to keep the feature, which I assume we are?)